apps/lenra/lib/lenra/kubernetes/api_services.ex

```diff
@@ -44,44 +45,19 @@ defmodule Lenra.Kubernetes.ApiServices do
 
     build_name = "build-#{service_name}-#{build_number}"
 
-    base64_repository = Base.encode64(app_repository)
-    base64_repository_branch = Base.encode64(app_repository_branch || "")
-
-    base64_callback_url = Base.encode64("#{runner_callback_url}/runner/builds/#{build_id}?secret=#{runner_secret}")
-
-    base64_image_name = Base.encode64(Apps.image_name(service_name, build_number))
-
-    secret_body =
-      Jason.encode!(%{
-        apiVersion: "v1",
-        kind: "Secret",
-        type: "Opaque",
-        metadata: %{
-          name: build_name,
-          namespace: kubernetes_build_namespace
-        },
-        data: %{
-          APP_REPOSITORY: base64_repository,
-          REPOSITORY_BRANCH: base64_repository_branch,
-          CALLBACK_URL: base64_callback_url,
-          IMAGE_NAME: base64_image_name
-        }
-      })
-
     secret_response =
-      Finch.build(:post, secrets_url, headers, secret_body)
-      |> Finch.request(PipelineHttp)
-      |> response(:secret)
+      create_k8s_secret(build_name, kubernetes_build_namespace, %{
+        APP_REPOSITORY: app_repository,
+        REPOSITORY_BRANCH: app_repository_branch || "",
+        CALLBACK_URL: "#{runner_callback_url}/runner/builds/#{build_id}?secret=#{runner_secret}",
+        IMAGE_NAME: Apps.image_name(service_name, build_number)
+      })
 
     case secret_response do
-      {:ok, _} ->
+      {:ok} ->
```

Comment:
You changed this case but it causes an error when trying to deploy an application. 

Here are the logs : 

```console
11:10:37.446 request_id=F7juFiPvtMtTt9AAAprh [info] POST /api/apps/389/builds
11:10:37.582 request_id=F7juFiPvtMtTt9AAAprh [info] Sent 500 in 135ms
11:10:37.583 [error] #PID<0.24822.0> running LenraWeb.Endpoint (connection #PID<0.24819.0>, stream id 3) terminated
Server: api.staging.lenra.io:80 (http)
Request: POST /api/apps/389/builds
** (exit) an exception was raised:
    ** (CaseClauseError) no case clause matching: {:ok, %{APP_REPOSITORY: "https://github.com/lenra-io/template-bun-js.git", CALLBACK_URL: "http://lenra-server.lenra.svc.cluster.local:4000/runner/builds/915?secret=sZWshq6h0RNO9T1GgUnzLmPpDkSkDAoukmd30mTuwQAGIHYIIVdl7VD2h305", IMAGE_NAME: "registry.gitlab.com/lenra/platform/lenra-ci/staging/dd14cd88-07d9-4711-9d4c-1a1d9d02b8b4:2", REPOSITORY_BRANCH: "secret"}}
        (lenra 0.1.0) lib/lenra/kubernetes/api_services.ex:56: Lenra.Kubernetes.ApiServices.create_pipeline/6
        (lenra 0.1.0) lib/lenra/apps.ex:266: Lenra.Apps.trigger_pipeline/3
        (lenra 0.1.0) lib/lenra/apps.ex:240: Lenra.Apps.create_build_and_deploy/3
        (lenra_web 0.1.0) lib/lenra_web/controllers/build_controller.ex:23: LenraWeb.BuildsController.create/2
        (lenra_web 0.1.0) lib/lenra_web/controllers/build_controller.ex:1: LenraWeb.BuildsController.action/2
        (lenra_web 0.1.0) lib/lenra_web/controllers/build_controller.ex:1: LenraWeb.BuildsController.phoenix_controller_pipeline/2
        (phoenix 1.6.16) lib/phoenix/router.ex:354: Phoenix.Router.__call__/2
        (lenra_web 0.1.0) lib/lenra_web/endpoint.ex:1: LenraWeb.Endpoint.plug_builder_call/2

```

Reply:
@shiipou I'm tagging you because I can't set this comment as blocking for the PR as I am the creator.